pallets/bootstrap/src/lib.rs

```diff
@@ -227,9 +230,10 @@ pub mod pallet {
 					) {
 						MintedLiquidity::<T>::put((liq_asset_id, issuance)); // W:1
 						if PromoteBootstrapPool::<T>::get() {
-							if !T::RewardsApi::promote_pool(liq_asset_id) {
-								log!(error, "pool already promoted!");
-							}
+							T::RewardsApi::update_pool_promotion(
+								liq_asset_id,
+								Some(T::DefaultBootstrapPromotedPoolWeight::get()),
```

Comment:
This was really an afterthought, since this feature (bootstrap pool promotion), didn't exist when the rewards v2 branch, which this feature was initially built on, was branched out from develop.
I didn't think it was worth the time to consider it much potentially pushing the delivery (due to possible migration required), since the pool promotion could easily be upgraded later on without much effect.

Reply:
ok, that should not delay the release so we can improve on that later


P.S. when bootstrap is finalized bootstrap schedule is purged so there would be no migration required even if you would modify https://github.com/mangata-finance/mangata-node/blob/develop/pallets/bootstrap/src/lib.rs#L333

Reply:
I would still have to write the migration just incase the upgrade is scheduled while a bootstrap in on going.